bee-network/bee-autopeering/src/peer/lists.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{peer_id::PeerId, Peer};
+
+use crate::{
+    discovery::manager::VERIFICATION_EXPIRATION_SECS,
+    time::{self, Timestamp},
+};
+
+use serde::{
+    de::{SeqAccess, Visitor},
+    ser::SerializeStruct,
+    Deserialize, Serialize,
+};
+
+use std::{
+    collections::{HashSet, VecDeque},
+    fmt,
+    sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard},
+};
+
+// Maximum number of peers that can be managed.
+const DEFAULT_MAX_MANAGED: usize = 1000;
+// Maximum number of peers kept in the replacement list.
+const DEFAULT_MAX_REPLACEMENTS: usize = 10;
+
+type ActivePeersListInner = PeerRing<ActivePeer, DEFAULT_MAX_MANAGED>;
+type ReplacementListInner = PeerRing<Peer, DEFAULT_MAX_REPLACEMENTS>;
+type MasterPeersListInner = HashSet<PeerId>;
+
+#[derive(Clone)]
+pub struct ActivePeer {
+    peer: Peer,
+    metrics: PeerMetrics,
+}
+
+impl ActivePeer {
+    pub(crate) fn new(peer: Peer) -> Self {
+        Self {
+            peer,
+            metrics: PeerMetrics::default(),
+        }
+    }
+
+    pub(crate) fn peer(&self) -> &Peer {
+        &self.peer
+    }
+
+    pub(crate) fn peer_mut(&mut self) -> &mut Peer {
+        &mut self.peer
+    }
+
+    pub(crate) fn peer_id(&self) -> &PeerId {
+        self.peer.peer_id()
+    }
+
+    pub(crate) fn metrics(&self) -> &PeerMetrics {
+        &self.metrics
+    }
+
+    pub(crate) fn metrics_mut(&mut self) -> &mut PeerMetrics {
+        &mut self.metrics
+    }
+
+    pub(crate) fn into_peer(self) -> Peer {
+        self.peer
+    }
+
+    pub(crate) fn to_vec(&self) -> Vec<u8> {
+        bincode::serialize(self).expect("serialization error")
+    }
+
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Self {
```

Comment:
I would harmonise either `to_vec`/`from_slice` or `to_bytes/from_bytes`

Reply:
Agree! Decided to go for the latter :smile: